sys/compat/freebsd64/freebsd64_uipc.c

```diff
@@ -287,14 +301,23 @@ freebsd64_copyin_control(struct mbuf **mp, char * __capability buf,
 		newlen += CMSG_SPACE(datalen);
 	}
 
-	if (newlen > MCLBYTES)
-		return (EINVAL);
-
-	m = m_get2(newlen, M_WAITOK, MT_CONTROL, 0);
-	m->m_len = newlen;
+	if (newlen <= MCLBYTES) {
+		m = m_get2(newlen, M_WAITOK, MT_CONTROL, 0);
+		m->m_len = newlen;
+		md = mtod(m, caddr_t);
+	} else {
+		m = m_get(M_WAITOK, MT_CONTROL);
+		m->m_ext.ext_buf = malloc(newlen, M_CMSG, M_WAITOK|M_ZERO);
```

Comment:
This needs to set `m->m_data` to `m->m_ext.ext_buf` and then you can leave the `mtod` below alone (that is, put it back at the start of the loop).  Otherwise `mtod` will still point to the internal data array in the mbuf which might be the source of your breakage.